inc/user.class.php

```diff
@@ -75,24 +75,35 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
    static private function getUserGroup($entity, $userid, $filter = '', $first = true) {
       global $DB;
 
-      $query = "SELECT glpi_groups.id
-                FROM glpi_groups_users
-                INNER JOIN glpi_groups ON (glpi_groups.id = glpi_groups_users.groups_id)
-                WHERE glpi_groups_users.users_id='$userid'".
-                getEntitiesRestrictRequest(' AND ', 'glpi_groups', '', $entity, true, true);
+      $criteria = [
+         'SELECT' => ['glpi_groups.id'],
+         'FROM'   => 'glpi_groups_users',
+         'INNER JOIN' => [
+            'glpi_groups' => [
+               'ON' => [
+                  'glpi_groups_users' => 'groups_id',
+                  'glpi_groups'       => 'id'
+               ]
+            ]
+         ],
+         'WHERE' => [
+            'glpi_groups_users.users_id' => $userid,
+             getEntitiesRestrictCriteria('glpi_groups', '', $entity, true, true)
+         ],
+         'ORDER' => ['glpi_groups_users.id']
+      ];
 
       if ($filter) {
-         $query .= "AND ($filter)";
+         $criteria['WHERE'][] = new QueryExpression($filter);
```

Comment:
Method is used only at 2 distinct places, please transform `$filter` into an array in order to also remove raw SQL in caller method.
```suggestion
         $criteria['WHERE'][] = $filter;
```